### PR TITLE
Enable Scalafmt and format everything

### PR DESCRIPTION
**Pull request by ckipp01**
This PR adds in 2 commits. One being a formatting commit since there is a `.scalafmt.conf` in this repo, but it was never enforced. The second commit adds a check in CI for formatting and also creates a `.git-blame-ignore-revs` file with the formatting commit in it.

### CODE REVIEW

**Review by lefou — CHANGES_REQUESTED** [review]
Can you please isolate the formatting commit even more. I'd like to have the tooling changes to enable the `ScalafmtModule` in `build.sc` not end up in `.git-blame-ignore-revs`.

**Comment by ckipp01** [comment]
> Can you please isolate the formatting commit even more. I'd like to have the tooling changes to enable the `ScalafmtModule` in `build.sc` not end up in `.git-blame-ignore-revs`.

Sure, no problem. I've adjusted it now so that the commit that is ignored _only_ has the actual formatting changes.

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Format the codebase for the first time
+6ae7a064eb53980dd391f6bd6484c12efe7083b4
```
Comment (lefou) [inline .git-blame-ignore-revs:2]:
```suggestion
b049841d5707d5bd87be516d8cda7be2a7585eae
```

**Review by lefou — CHANGES_REQUESTED** [review]
The `.git-blame-ignore-revs` still references an outdated commit. Beside that, this PR looks good to me.

File: .git-blame-ignore-revs (hunk 1 above)
Comment (ckipp01) [reply .git-blame-ignore-revs]: Ah dangit, totally forgot to update it after I rebased. Thanks! Fixed.

**Review by lefou — APPROVED** [review]
Looks good to me. Thank you!

**Pull request merged**